### PR TITLE
create ci-kubernetes-build-fast canary job to validate job running in the cluster

### PR DESCRIPTION
**Pull request by cpanato**
create duplicate `ci-kubernetes-build-fast` to run in the `k8s-infra-prow-build` cluster to validate the execution build

ref: https://github.com/kubernetes/test-infra/issues/19484 and https://github.com/kubernetes/kubernetes/issues/95173

/cc @spiffxp Dont know if there is anything else that we need to change, but this is an initial PR, let me know if this is correct.

### CODE REVIEW

**Review by spiffxp — CHANGES_REQUESTED** [review]
This needs to be modified to push to gs://k8s-release-dev instead of gs://kubernetes-release-dev

I'm not sure off the top of my head which flag needs to be set where for this

File: config/jobs/kubernetes/sig-release/kubernetes-builds.yaml
Comment (spiffxp) [inline config/jobs/kubernetes/sig-release/kubernetes-builds.yaml:138]: I would suggest leaving it off of blocking until we've gained confidence it's working as intended
```suggestion
    testgrid-dashboards: sig-release-master-informing, sig-testing-canaries
```

File: config/jobs/kubernetes/sig-release/kubernetes-builds.yaml
Comment (spiffxp) [inline config/jobs/kubernetes/sig-release/kubernetes-builds.yaml:141]:
```suggestion
    description: 'Ends up running: make quick-release. Canary: runs in k8s-infra-prow-build, pushes to gs://k8s-release-dev'
```

**Review by spiffxp — CHANGES_REQUESTED** [review]
Some suggested changes as well

**Comment by cpanato** [comment]
@spiffxp thanks for your review and input! i will check how to change that and apply here.

**Comment by cpanato** [comment]
@spiffxp done, PTAL

File: config/jobs/kubernetes/sig-release/kubernetes-builds.yaml
Comment (spiffxp) [inline config/jobs/kubernetes/sig-release/kubernetes-builds.yaml:124]: I'll merge this as-is just to see if the job runs successfully, but for followup this should be different bucket / same path.  I think what you have here is different bucket / different path

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
I'm going to assume you've got the flag correct.  This will need changes before we can start moving forward, but I'll merge this to get early signal.

File: config/jobs/kubernetes/sig-release/kubernetes-builds.yaml
Comment (cpanato) [reply config/jobs/kubernetes/sig-release/kubernetes-builds.yaml:124]: ok, followup PR: https://github.com/kubernetes/test-infra/pull/19518

**Pull request merged**
